packages/backend/Dockerfile

```diff
@@ -1,16 +1,20 @@
-FROM node:14-slim as builder
+FROM node:16-slim as builder
 
 WORKDIR /workspace
 
 RUN apt update
 RUN apt install -y curl xz-utils python3 build-essential
-COPY package.json yarn.lock tsconfig.build.json .yarnrc.yml ./
-COPY .yarn ./.yarn
-COPY ./packages ./packages
-COPY ./mock-servers ./mock-servers
 
-RUN yarn install --immutable
-RUN yarn workspaces foreach -ptvR --from backend run build \
-  && yarn workspace backend build
+# version in curl is not the version used. Dependent on the last command
+RUN curl -f https://get.pnpm.io/v6.16.js | node - add --global pnpm@7.9.3
+
+# pnpm fetch does require only lockfile
+COPY pnpm-lock.yaml ./
+RUN pnpm fetch
+
+ADD . ./
```

Comment:
nit: should we add `.prettierrc.yml`, `.prettierignore`, infrastructure etc to the `.dockerignore` ?